src/pulsar_producer.erl

```diff
@@ -16,6 +16,7 @@
 
 -behaviour(gen_statem).
 
+-include("src/pulsar_producer_internal.hrl").
```

Comment:
better to move this file to `include` dir.
include dir is not necessary always for external use.

Reply:
:heavy_check_mark: 

I thought that it was a common convention to put the internal headers in `src/`. Moved to `include/`.